appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -651,10 +662,14 @@ public PayaraMicroImpl addDeployment(String pathToWar) {
     public PayaraMicroImpl addDeploymentFile(File file) {
         //if (runtime != null) {
         checkNotRunning();
-        if (deployments == null) {
-            deployments = new LinkedList<>();
+        if ((file.isFile() && !hasJavaArchiveExtension(file.getName()))
```

Comment:
This looks like duplication of `DeploymentFileValidator`

Reply:
All validators now have package-visible validate method. Can I change visibility?

Reply:
Either that, or since they have no state, moving the validation into `static` method.

Reply:
May be change visibility of the `validate` method of `RUNTIME_OPTION` enum as a single point of changes?